File: docs/guides/python/setSettingsThenSaveObjects.py
from algoliasearch.search.client import SearchClientSync

from algoliasearch.search.models.index_settings import IndexSettings


def _get_app_id_for(_user):
    # Implement your own logic here
    return ""


def _get_indexing_api_key_for(_user):
    # Implement your own logic here
    return ""


playlists = []  # Your records

for playlist in playlists:
    app_id = _get_app_id_for(playlist["user"])
    api_key = _get_indexing_api_key_for(playlist["user"])

    _client = SearchClientSync(app_id, api_key)
    settings = IndexSettings(attributes_for_faceting=["filterOnly(user)"])
    try:
        _client.set_settings(
            index_name="<YOUR_INDEX_NAME>",
            index_settings=settings,
        )

        _client.save_objects(
            index_name="<YOUR_INDEX_NAME>",
            objects=playlists,
        )
    except Exception as e:
        print(f"Error: {e}")

File: docs/guides/go/src/setSettingsThenSaveObjects.go
package main

import (
	"context"
	"fmt"

	"github.com/algolia/algoliasearch-client-go/v4/algolia/next/search"
)

func getAppIDFor(_ string) (string, error) {
	return "", nil // Implement your logic here
}

func getIndexingAPIKeyFor(_ string) (string, error) {
	return "", nil // Implement your logic here
}

func setSettingsThenSaveObjects() {
	playlists := []map[string]any{{ /* Your records */ }}

	for _, playlist := range playlists {
		// Fetch from your own data storage and with your own code
		// the associated application ID and API key for this user
		appID, err := getAppIDFor(playlist["user"].(string))
		if err != nil {
			fmt.Println(err)

			return
		}

		apiKey, err := getIndexingAPIKeyFor(playlist["user"].(string))
		if err != nil {
			fmt.Println(err)

			return
		}

		client, err := search.NewClient(appID, apiKey)
		if err != nil {
			fmt.Println(err)

			return
		}

		settings := &search.IndexSettings{
			AttributesForFaceting: []string{"filterOnly(user)"},
		}

		_, err = client.SetSettings(context.Background(), "<YOUR_INDEX_NAME>", settings, nil)
		if err != nil {
			panic(err)
		}

		_, err = client.SaveObjects(context.Background(), "<YOUR_INDEX_NAME>", playlists)
		if err != nil {
			panic(err)
		}
	}
}
